examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt

```diff
@@ -0,0 +1,429 @@
+                        .coordinates(originLocation.toPoint(), null, latLng.toPoint())
+                        .alternatives(true)
+                        .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                        .build(),
+                    routesReqCallback
+                )
+            }
+            true
+        }
+
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            locationComponent = mapboxMap.locationComponent.apply {
+                activateLocationComponent(
+                    LocationComponentActivationOptions.builder(this@InstructionViewActivity, style)
+                        .build()
+                )
+                cameraMode = CameraMode.TRACKING
+                isLocationComponentEnabled = true
+            }
+
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap).also {
+                it.addProgressChangeListener(mapboxNavigation)
+                it.setCamera(DynamicCamera(mapboxMap))
+            }
+        }
+    }
+
+    // InstructionView Feedback Bottom Sheet listener
+    override fun onFeedbackSelected(feedbackItem: FeedbackItem?) {
+        feedbackItem?.let { feedback ->
+            mapboxMap?.snapshot { snapshot ->
+                MapboxNavigation.postUserFeedback(
+                    TelemetryUtils.obtainUniversalUniqueIdentifier(),
+                    feedback.feedbackType, feedback.description, encodeSnapshot(snapshot)
+                )
+            }
+        }
+    }
+
+    override fun onFeedbackDismissed() {
+        // do nothing
+    }
+
+    private fun initNavigation() {
+        val accessToken = Utils.getMapboxAccessToken(this)
+        mapboxNavigation = if (Utils.shouldSimulateRoute(applicationContext)) {
+            Timber.d("initNavigation simulate route")
+            MapboxNavigation(
+                applicationContext,
+                accessToken,
+                MapboxNavigation.defaultNavigationOptions(this, accessToken),
+                replayRouteLocationEngine
+            )
+        } else {
+            Timber.d("initNavigation location engine")
+            MapboxNavigation(
+                applicationContext,
+                accessToken,
+                MapboxNavigation.defaultNavigationOptions(this, accessToken)
+            )
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initListeners() {
+        startNavigation.setOnClickListener {
+            Timber.d("start navigation")
+            route?.let {
+                if (Utils.shouldSimulateRoute(applicationContext)) {
+                    replayRouteLocationEngine.assign(it)
+                }
+
+                // navigationMapboxMap.updateLocationLayerRenderMode(RenderMode.GPS)
+                // navigationMapboxMap.updateCameraTrackingMode(NavigationCamera.NAVIGATION_TRACKING_MODE_GPS)
+                // navigationMapboxMap.startCamera(it)
```

Comment:
Let's uncomment these lines. It's good experience to have the camera navigate with the route